recipes-core/rauc/files/rauc-mark-good

```diff
@@ -110,12 +130,20 @@ if $FIRST_BOOT; then
     fi
 fi
 
+USB_BOOT=false
+if usb_boot; then
+    USB_BOOT=true
+fi
+
 # Mark the current boot good
 # Suppress errors when running on non-A/B capable firmware/boot loader
-if $HAS_NIBOOT_EFI; then
-    rauc status mark-good >/dev/null
-else
-    rauc status mark-good >/dev/null || true
+# Skip call to mark-good if this boot a one-time USB boot
+if ! $USB_BOOT; then
+    if $HAS_NIBOOT_EFI; then
+        rauc status mark-good >/dev/null
+    else
+        rauc status mark-good >/dev/null || true
+    fi
 fi
```

Comment:
Could you move the status messages from the bottom of the file into this structure.
1. They act as additional documentation.
2. If someone is grepping the file because they noticed the status output, they won't have to go digging back through the code to figure out *why* the messages were generated.

Reply:
Did this one too.